cafebabel/tests/test_translations.py

```diff
@@ -0,0 +1,237 @@
+        data=data)
+    assert response.status_code == HTTPStatus.NOT_FOUND
+
+
+def test_translation_access_draft_should_return_200(client, translation):
+    response = client.get(f'/article/translation/{translation.id}/')
+    assert response.status_code == HTTPStatus.OK
+
+
+def test_translation_access_have_original_article_link(client, translation):
+    response = client.get(f'/article/translation/{translation.id}/')
+    text = response.get_data(as_text=True)
+    assert ((f'Translated from '
+             f'<a href="/article/draft/{translation.original_article.id}/">'
+             f'article title')
+            in text)
+
+
+def test_translation_access_have_translator(client, translation):
+    response = client.get(f'/article/translation/{translation.id}/')
+    text = response.get_data(as_text=True)
+    assert f'by {translation.translator}.' in text
+
+
+def test_translation_access_published_should_return_404(client, translation):
+    translation.status = 'published'
+    translation.save()
+    response = client.get(f'/article/translation/{translation.id}/')
+    assert response.status_code == HTTPStatus.NOT_FOUND
+
+
+def test_translation_access_wrong_id_should_return_404(client, translation):
+    response = client.get(f'/article/translation/foo{translation.id}/')
+    assert response.status_code == HTTPStatus.NOT_FOUND
+
+
+def test_translation_update_should_display_form(client, user, translation):
+    login(client, user.email, 'secret')
+    response = client.get(f'/article/translation/{translation.id}/edit/')
+    assert response.status_code == HTTPStatus.OK
+    text = response.get_data(as_text=True)
+    assert '<textarea id=body name=body required>body text</textarea>' in text
+
+
+def test_translation_update_requires_login(client, translation):
+    response = client.get(f'/article/translation/{translation.id}/edit/')
+    assert response.status_code == HTTPStatus.FOUND
+    assert ('/login?next=%2Farticle%2Ftranslation%2F'
+            in response.headers.get('Location'))
+
+
+def test_translation_update_values_should_redirect(client, user, translation):
+    login(client, user.email, 'secret')
+    data = {
+        'title': 'Modified title',
+        'summary': 'Modified summary',
+        'body': 'Modified body',
+    }
+    response = client.post(
+        f'/article/translation/{translation.id}/edit/', data=data)
+    assert response.status_code == HTTPStatus.FOUND
+    translation = Translation.objects.first()
+    assert (response.headers.get('Location') ==
+            f'http://localhost/article/translation/{translation.id}/')
+    assert translation.title == 'Modified title'
+    assert (get_flashed_messages() ==
+            ['Your translation was successfully updated.'])
+
+
+def test_translation_update_empty_summary(client, user, translation):
+    login(client, user.email, 'secret')
+    data = {
+        'title': translation.title,
+        'summary': '',
```

Comment:
Wasn't this mandatory in the end?

Reply:
Indeed, actually the test pass because the value when empty is not updated. I remove it.